app/src/main/kotlin/com/wire/android/ui/userprofile/self/SelfUserProfileViewModel.kt

```diff
@@ -124,7 +124,14 @@ class SelfUserProfileViewModel @Inject constructor(
 
     fun addAccount() {
         viewModelScope.launch {
-            navigationManager.navigate(NavigationCommand(NavigationItem.CreatePersonalAccount.getRouteWithArgs()))
+            val selfServerLinks = with(selfServerLinks()) {
+                when (this) {
+                    is SelfServerConfigUseCase.Result.Failure -> return@launch
+                    is SelfServerConfigUseCase.Result.Success -> this.serverLinks
+                }
+            }
```

Comment:
I think it would be easier to just do it like this : 

```
val selfServerLinks = selfSelverLInks() 

if(selfServerLinks is SelfServerConfigUseCase.Result.Failure) cancel()

```
with kind of defeats the purpose here in my opinion